### ISSUE TITLE
Always uses main branch

### ISSUE THREAD

**Issue by botahamec** [rust-lang/rust-analyzer#13487]
I have this in my Cargo.toml

```toml
[dependencies]
# ...
texture_packer = { git = "https://github.com/botahamec/piston_texture_packer", branch = "u16" }
```

But I'm still getting a compiler error which should only appear on the main branch. I've tried restarting rust-analyzer, restarting VSCode, and running `cargo update`. None of that seems to fix it.

**rust-analyzer version**: rust-analyzer version: 0.4.1259-standalone (53b6d69e9 2022-10-24)

**rustc version**: 1.66.0-nightly (dcb376115 2022-10-20)

**Comment by jhgg** [comment]
Hmm. Pulling down the repo and what not is still handled by cargo - so if there was a bug here it would most likely be in cargo. 

Additionally, rust-analyzer does not emit compiler errors beyond the output of cargo check (or whatever fly check command you specify in config.).

**Comment by flodiebold** [comment]
What does running `cargo check` show?

**Comment by lnicola** [comment]
Or `cargo check --all-targets`.

This usually happens when you switch branches while `cargo check` is running.

**Comment by botahamec** [comment]
> Additionally, rust-analyzer does not emit compiler errors beyond the output of cargo check
@jhgg I'm talking about errors that show up in the editor. There were errors shown to me in rust-analyzer that weren't being shown by `cargo check`.

> What does running cargo check show?
@flodiebold I already ran `cargo clean`, but even before that it was giving me no errors whatsoever.

**Comment by lnicola** [comment]
> There were errors shown to me in rust-analyzer that weren't being shown by cargo check.

That's because we run `cargo check --all-targets`, which can show different errors. Did you switch branches immediately after loading the project?

**Comment by botahamec** [comment]
I don't think the check was running when I switched branches, but I'm not sure. I can say that the errors were not in a test, and that the error wouldn't be caused by a feature. The error was saying that it was expecting a `u8` instead of a `u16`. My branch was made so that the `u16` was needed instead.
